mlrun/db/httpdb.py

```diff
@@ -140,6 +153,54 @@ def get_base_api_url(self, path: str, version: str = None) -> str:
         url = f"{self.base_url}/{path_prefix}/{path}"
         return url
 
+    def request_with_retry(self, method, url, **kwargs):
+        max_retries = (
+            HTTP_RETRY_AMOUNT
+            if config.httpdb.retry_api_call_on_exception == "enabled"
+            else 1
+        )
+        retry_count = 0
+        while True:
+            try:
+                response = self.session.request(method, url, **kwargs)
+                return response
+            except tuple(HTTP_RETRY_EXCEPTIONS.keys()) as exc:
+                retry_count += 1
+                if retry_count >= max_retries:
```

Comment:
One way to fix the confusion is to change this to `>`.

Reply:
Changed it by moving the `retry_count += 1` to the end. As it was indeed counting the tries and not the retries